src/main/java/co/elastic/logstash/filters/elasticintegration/EventProcessorBuilder.java

```diff
@@ -308,17 +311,33 @@ public synchronized EventProcessor build(final PluginContext pluginContext) {
     }
 
     private static ScriptService initScriptService(final Settings settings, final ThreadPool threadPool) {
-        final List<Whitelist> painlessBaseWhitelist = getPainlessBaseWhiteList();
-        final Map<ScriptContext<?>, List<Whitelist>> scriptContexts = Map.of(
-                IngestScript.CONTEXT, painlessBaseWhitelist,
-                IngestConditionalScript.CONTEXT, painlessBaseWhitelist);
-
         Map<String, ScriptEngine> engines = new HashMap<>();
-        engines.put(PainlessScriptEngine.NAME, new PainlessScriptEngine(settings, scriptContexts));
+        engines.put(PainlessScriptEngine.NAME, getPainlessScriptEngine(settings));
         engines.put(MustacheScriptEngine.NAME, new MustacheScriptEngine());
         return new ScriptService(settings, engines, ScriptModule.CORE_CONTEXTS, threadPool::absoluteTimeInMillis);
     }
 
+    private static ScriptEngine getPainlessScriptEngine(final Settings settings) {
+        try (PainlessPlugin painlessPlugin = new PainlessPlugin()) {
+
+            painlessPlugin.loadExtensions(new ExtensiblePlugin.ExtensionLoader() {
+                @Override
+                @SuppressWarnings("unchecked")
+                public <T> List<T> loadExtensions(Class<T> extensionPointType) {
+                    if (extensionPointType == PainlessExtension.class) {
+                        return List.of((T) new ProcessorsWhitelistExtension());
+                    } else {
+                        return List.of();
+                    }
+                }
+            });
```

Comment:
review note: a single-use `ExtensiblePlugin.ExtensionLoader` is common-place in Elasticsearch core and looks very similar to this.

While we know that `PainlessPlugin#loadExtensions(ExtensiblePlugin.ExtensionLoader)` will only ever send `ExtensiblePlugin.ExtensionLoader#loadExtensions(PainlessExtension.class)`, we need to accept any class in order to satisfy the interface.